aries_cloudagent/askar/store.py

```diff
@@ -156,16 +168,26 @@ async def open_store(self, provision: bool = False) -> "AskarOpenStore":
                     self.key_derivation_method,
                     self.key,
                 )
+                if self.rekey:
+                    await Store.rekey(store, self.key_derivation_method, self.rekey)
```

Comment:
Is the key derivation method something that we would want to expose, or using the currently "saved" one is the only feasible option?

Reply:
This is from the config value. Currently it uses the same value. Meaning you need to use the same method for the rekey as the original key method. I saw a config that doesn't exist commented out that would allow the rekey to use a different method.

Reply:
It it isn't provided it defaults to `kdf:argon2i:mod`

Reply:
Perhaps we could consider adding wallet.rekey_derivation_method to allow for simultaneous changes to the KDF.

Reply:
Yes. I think that's a good idea.